0012-algorithms-migration.md

```diff
@@ -0,0 +1,178 @@
+# Plan to move `Qiskit/qiskit-terra/algorithms` module to `Qiskit-community/qiskit-algorithms` independent repo
+
+| **Status**        | **Proposed**                                                                              |
+|:------------------|:------------------------------------------------------------------------------------------|
+| **RFC #**         | ####                                                                                      |
+| **Authors**       | [Elena Peña Tapia](https://github.com/ElePT), [Steve Wood](https://github.com/woodsp-ibm) |
+| **Submitted**     | 2023-07-05                                                                                |
+| **Updated**       | YYYY-MM-DD                                                                                |
+
+
+## Summary
+
+This RFC was created to document the discussions around the future of the `qiskit.algorithms` module.
+This module provides a series of interfaces and implementations of commonly used quantum algorithms based on 
+Qiskit's core components. Historically, the module was created as part of Qiskit Aqua, together with the now
+community-based application modules (Nature, Machine Learning, Optimization, Finance). It currently "lives"
+in `qiskit-terra`, and over the past year has been refactored to use the Qiskit Primitives and sustain the 
+Opflow and Quantum Instance deprecations.
+
+The content of `qiskit.algorithms` is heterogeneous. Some components are algorithm implementations, while others
+are "sub-routines" or algorithm utilities, such as gradients, optimizers or state fidelities. Some of these
+utilities are implementatons from scratch, others are wrappers over commonly used libraries (such as the 
+SciPy optimizers).
+
+With Qiskit moving towards a leaner definition, we are now revisiting the future of this module. 
+We would like to tell apart the fundamental components for algorithm design from algorithm implementations, and 
+find a better suiting location for both of them while minimizing the disruption caused to users.
```

Comment:
```suggestion
With Qiskit moving towards a leaner definition and focusing on primitives as quantum hardware access model, the future of `qiskit.algorithms` is being revisited. The classes in the module tend to hide the primitives interfaces and serve as convenience wrappers for abstractions, beyond the current definition of Qiskit.
As such, they do provide value to users to it would be important to 
find a better suiting location for them while minimizing the disruption caused to users.
```